# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../qt4/qt4-translations-build-fix.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

without this patch we get following error when linking
-----------------------------------------------------------------------------
g++ -Wl,-rpath-link,/TOOLCHAIN/src.qt4.uma.20130618.102156.21560.builder64/qt-everywhere-opensource-src-4.8.4/lib -Wl,-O1 -o qm_phony_target
/usr/lib/gcc/i686-unknown-linux-gnu/4.7.3/../../../crt1.o: In function `_start':
(.text+0x18): undefined reference to `main'
collect2: error: ld returned 1 exit status
-----------------------------------------------------------------------------

--- qt-everywhere-opensource-src-4.8.4/translations/translations.pro.orig	2013-06-18 12:51:39.085600311 +0200
+++ qt-everywhere-opensource-src-4.8.4/translations/translations.pro	2013-06-18 12:53:40.134783556 +0200
@@ -101,7 +101,7 @@
 updateqm.CONFIG += no_link
 QMAKE_EXTRA_COMPILERS += updateqm
 
-isEmpty(vcproj) {
+!isEmpty(vcproj) {
     QMAKE_LINK = @: IGNORE THIS LINE
     OBJECTS_DIR =
     win32:CONFIG -= embed_manifest_exe
@@ -111,7 +111,7 @@
     phony_src.input = PHONY_DEPS
     phony_src.output = phony.c
     phony_src.variable_out = GENERATED_SOURCES
-    phony_src.commands = echo int main() { return 0; } > phony.c
+    phony_src.commands = echo \"int main() { return 0; }\" > phony.c
     phony_src.name = CREATE phony.c
     phony_src.CONFIG += combine
     QMAKE_EXTRA_COMPILERS += phony_src
